labels-config-service-impl/src/main/java/org/hypertrace/label/config/service/LabelsConfigServiceImpl.java

```diff
@@ -230,19 +195,21 @@ public void updateLabel(
     try {
       LabelData updateLabelData = request.getData();
       RequestContext requestContext = RequestContext.CURRENT.get();
-      if (systemLabelsIdLabelMap.containsKey(request.getId())) {
-        // Updating a system label will error
-        responseObserver.onError(new StatusRuntimeException(Status.INVALID_ARGUMENT));
-        return;
-      }
+
       if (isDuplicateKey(requestContext, request.getId(), updateLabelData.getKey())) {
         responseObserver.onError(new StatusRuntimeException(Status.ALREADY_EXISTS));
         return;
       }
       Label oldLabel =
           labelStore
               .getData(requestContext, request.getId())
-              .orElseThrow(Status.NOT_FOUND::asRuntimeException);
+              .orElseGet(
+                  () ->
+                      Optional.ofNullable(
+                              systemLabelInfoUtils
+                                  .getSystemLabelsIdLabelMap(getExistingLabels(requestContext))
```

Comment:
i am confused here? why do you need to find system labels ids from existing labels? We dont change system config ids when we store them in DB

Reply:
It should be as simple as

```
Label label =
          labelStore
              .getData(requestContext, labelId)
              .orElseGet(
                  () ->
                      Optional.ofNullable(
                              systemLabelInfoUtils
                                  .getDefaultSystemLabels()
                                  .filter(label -> label.getId().equals(labelId))
                                  .findFirst()
                          .orElseThrow(Status.NOT_FOUND::asRuntimeException));
```